### ISSUE TITLE
The structure of the igGrid data table is not semantically correct

### ISSUE THREAD

**Issue by kchanda24** [IgniteUI/ignite-ui#2248]
## Description  
 During a recent evaluation by a third-party accessibility vendor, it we found out that in the data table of the igGrid the screen reader reads table header elements as one table and its data cells as a different table.This is not aligned with WCAG 2.0 Level A guidelines. The corresponding WCAG success criteria is 1.3.1 - Info and Relationships.

## Version and Browser used : 
 * ignite-ui version: any
 * browser: any

## Steps to reproduce  

1. Go to this [sample](https://jsfiddle.net/georgianastasov/kn7ru52s/2/).
2. Use NVDA or other accessibility tool.
3. Reach the data table
4. Tab though the headers and data

## Result  
The screen reader reads table header elements as one table and its data cells as a different table.

## Expected result  
What is the expected result after following the steps to reproduce?
The screen reader should read the data and headers as a single table and the relationships between header and elements should be proper.

When data tables are not coded according to standards, screen reader users will find it difficult to understand the data relationships.

If there is no way to combine the tables into one their should be a way to define relationships such that assistive technologies can understand the tables correctly.

## Resources

Accessible tables:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table#Accessibility_concerns
https://www.w3.org/WAI/ARIA/apg/patterns/grid/examples/data-grids/

**Comment by ChronosSF** [comment]
@kchanda24 , to add a bit more information here, there are relationships built-in in the grid that links cells and headers even if it's not exactly the same as having them in the same table element. The most simple example is that each cell has `aria-describedby` attribute that points to the `th` element it belongs to. This allows screen readers such as NVDA to read the column header when reading the contents of the cell. Going through examples of modern accessible fixed header tables (as this separation was really only needed for fixed header grids) I don't really see NVDA working with them any better than with igGrid. If you could provide more concrete examples of what is lacking on the NVDA reading, we could try to provide solutions for them on a case-by-case basis.

**Comment by kchanda24** [comment]
@ChronosSF It is difficult for us and unnecessary effort from our side to find out examples which work as expected. This is something that we are completely dependent on Infragistics. We are merely forwarding our requirements from an accessibility perspective.

**Comment by ChronosSF** [comment]
@kchanda24 , I understand this. However, as @georgianastasov already mentioned the change that we have to make is simply unfeasible as it'll be tremendous effort to redo the grid rendering. Especially when it's more or less only to appease a tool that reports an issue based on set amount of rules without actually testing the readout on NVDA (and other narrators). What I am trying to suggest is that your users would not have a worse experience when using igGrid because we don't follow this rule to the letter. Testing how cells and columns are read between our approach and a grid having the thead and tbody in the same table are very similar, only slightly differing in the order of how the different parts are read.
